File: 566.cpp
#include"mod.h"
class Solution {
public:
    vector<vector<int>> matrixReshape(vector<vector<int>>& nums, int r, int c) {
        int tot_num=r*c;
        int cur_num=nums.size()*nums[0].size();
        if(tot_num!=cur_num)return nums;
        vector<vector<int>>ret(r, vector<int>(c, 0));
        int ret_x=0, ret_y=0;
        for(int x=0; x<nums.size(); x++){
          for(int y=0; y<nums[x].size(); y++){
            ret[ret_x][ret_y]=nums[x][y];
            ++ret_y;
            if(ret_y>=c){
              ret_y=0;
              ++ret_x;
            }
          }
        }
      return ret;
    }
};

File: py/566.py
class Solution(object):
    def matrixReshape(self, nums, r, c):
        """
        :type nums: List[List[int]]
        :type r: int
        :type c: int
        :rtype: List[List[int]]
        """
        tot_num=r*c;
        cur_num=len(nums)*len(nums[0]);
        if tot_num!= cur_num:
            return nums;
        ret=[[]];
        ret_x=0;
        for x in range(len(nums)):
            for y in range(len(nums[x])):
                ret[ret_x].append(nums[x][y])
                if len(ret[ret_x])==c:
                    ret.append([]);
                    ret_x=ret_x+1;
        x=len(ret);
        if ret[x-1]==[]:
            del ret[x-1];
        return ret;

        